src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -901,6 +901,74 @@ BOOST_AUTO_TEST_CASE(GradientFastLSTMLayerTest)
   BOOST_REQUIRE_LE(CheckGradient(function), 0.2);
 }
 
+/**
+ * Testting the overloaded Forward of LSTM Layer.
+ * Besides output, the overloaded function
+ * provides access to cell state of LSTM.
+ */
+BOOST_AUTO_TEST_CASE(CellStateParamLSTMLayerTest)
+{
+  const size_t rho = 5, inputSize = 3, outputSize = 2;
+
+  // Provide input of all ones.
+  arma::cube input = arma::ones(inputSize, outputSize, rho);
+
+  arma::mat inputGate, forgetGate, outputGate, hidden;
+  arma::mat out_lstm, cell_lstm;
+
+  // LSTM layer.
+  LSTM<> lstm(inputSize, outputSize, rho);
+  lstm.Reset();
+  lstm.ResetCell(rho);
+
+  // Initialize the weights to all ones.
+  lstm.Parameters().ones();
+
+  arma::mat inputWeight = arma::ones(outputSize, inputSize);
+  arma::mat outputWeight = arma::ones(outputSize, outputSize);
+  arma::mat bias = arma::ones(outputSize, input.n_cols);
+  arma::mat cell_calc = arma::zeros(outputSize, input.n_cols);
+  arma::mat out_calc = arma::zeros(outputSize, input.n_cols);
+
+  for (size_t seqNum = 0; seqNum < rho; ++seqNum)
+  {
+      // Wrap a matrix around our data to avoid a copy.
+      arma::mat stepData(input.slice(seqNum).memptr(),
+          input.n_rows, input.n_cols, false, true);
+
+      // Apply Forward on LSTM layer.
+      lstm.Forward(std::move(stepData),
+                   std::move(out_lstm),
+                   std::move(cell_lstm));
+
+      // Compute the value of cell state and output.
+      // i = sigmoid(W.dot(x) + W.dot(h) + W.dot(c) + b)
```

Comment:
Just curious, does this come from some python/numpy code?

Reply:
Used this one.
https://github.com/mlpack/mlpack/blob/7942ffc43c54570ff4b263c6f0d55f03352f5c63/src/mlpack/methods/ann/layer/lstm.hpp#L31